src/api/sapaicore.ts

```diff
@@ -76,7 +83,9 @@ export class SapAiCoreHandler implements ApiHandler {
 	}
 
 	private async getDeploymentForModel(modelId: string): Promise<string> {
-		if (!this.deployments) {
+
+		// If deployments are not fetched yet or the model is not found in the fetched deployments, fetch deployments
```

Comment:
We should still refresh just in case someone deletes and re-adds the model. In this case the ID would be different but the model name is the same.

But it is ok for now :)